### PR TITLE
Removed the unecessary "new" keyword from the docs

### PR DESCRIPTION
**Pull request by feinstein**
## Description

As the new Dart versions allow us to omit the `new` keyword on constructors and most docs and examples follow this, I think its reasonable to remove those old references from the docs.

## Related Issues

I took the liberty to not opening an issue for something as simple as this, but if it's needed, please let me know and I will open one.

## Tests

I added the following tests:

*Replace this with a list of the tests that you added as part of this PR. A change in behaviour with no test covering it
will likely get reverted accidentally sooner or later. PRs must include tests for all changed/updated/fixed behaviors. See [Test Coverage].*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/widgets/scroll_view.dart [hunk 1]
```diff
@@ -415,11 +415,11 @@ abstract class ScrollView extends StatelessWidget {
 ///
 /// This semantic index is not necessarily the same as the index of the widget in
 /// the scrollable, because some widgets may not contribute semantic
-/// information. Consider a [new ListView.separated()]: every other widget is a
+/// information. Consider a [ListView.separated()]: every other widget is a
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/scroll_view.dart]: while you're in here: remove the `()` here and below?

**Comment by goderbauer** [comment]
Looks like the analyzer has detected some trailing whitespace:

https://github.com/flutter/flutter/pull/51075/checks?check_run_id=456302126

Can you please remove those? Thanks!

**Comment by feinstein** [comment]
@goderbauer done.

File: packages/flutter/lib/src/widgets/scroll_view.dart (hunk 1 above)
Comment (goderbauer) [reply packages/flutter/lib/src/widgets/scroll_view.dart]: What about this one?

Comment (feinstein) [reply packages/flutter/lib/src/widgets/scroll_view.dart]: Oh, didn't see this comment, my apologies.

Comment (feinstein) [reply packages/flutter/lib/src/widgets/scroll_view.dart]: done

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**